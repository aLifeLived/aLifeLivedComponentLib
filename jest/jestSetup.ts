/* eslint-disable no-undef */
jest.useFakeTimers({ legacyFakeTimers: true }); // Stops reference error from appearing
