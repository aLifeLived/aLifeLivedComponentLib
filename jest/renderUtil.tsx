import React from 'react';
import { ThemeProvider } from '../src/themes/ThemeProvider';
import { render as rtlRender } from '@testing-library/react-native';
import { library } from '@fortawesome/fontawesome-svg-core';
import { iconList } from '../src/themes/systemIcon/IconList';

library.add(iconList);

function render(ui: any, { ...renderOptions } = {}) {
  function Wrapper({ children }: { children: any }) {
    return <ThemeProvider>{children}</ThemeProvider>;
  }

  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

// re-export everything for convenience
export * from '@testing-library/react-native';

// override render method
export { render };
