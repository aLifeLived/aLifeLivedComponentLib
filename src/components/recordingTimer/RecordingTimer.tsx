import React from 'react';

// Animation
import * as Animatable from 'react-native-animatable';

// Components
import { Box } from '../../themes/box/Box';
import { Text } from '../../themes/text/Text';

interface RecordingTimerProps {
  timerSeconds: string;
  isRecording: boolean;
  status: string;
}

export const RecordingTimer: React.FC<RecordingTimerProps> = ({
  timerSeconds,
  isRecording,
  status,
}) => {
  return (
    <Box>
      <Text textAlign="center" variant="intro">
        {timerSeconds}
      </Text>
      <Box flexDirection="row" alignItems="center">
        {isRecording ? (
          <Animatable.View animation="fadeIn" iterationCount={'infinite'}>
            <Box
              backgroundColor="backgroundError"
              width={10}
              height={10}
              borderRadius="medium"
              marginRight="s"
            />
          </Animatable.View>
        ) : null}
        <Text textAlign="center" variant="body">
          {status}
        </Text>
      </Box>
    </Box>
  );
};
