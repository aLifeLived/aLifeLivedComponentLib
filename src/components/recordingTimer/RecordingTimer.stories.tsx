import React from 'react';
import { storiesOf } from '@storybook/react-native';

// Component
import { RecordingTimer } from './RecordingTimer';
import CenterView from '../centerView/CenterView';

storiesOf('Recording timer', module)
  .addDecorator(Story => <CenterView children={Story()} />)
  .add('Loading state', () => (
    <RecordingTimer
      timerSeconds="0:00"
      isRecording={false}
      status="Loading please wait..."
    />
  ))
  .add('Ready state', () => (
    <RecordingTimer timerSeconds="0:00" isRecording={false} status="Ready" />
  ))
  .add('Buffering state', () => (
    <RecordingTimer
      timerSeconds="0:00"
      isRecording={false}
      status="Buffering..."
    />
  ))
  .add('Recording state', () => (
    <RecordingTimer timerSeconds="5:30" isRecording={true} status="Recording" />
  ))
  .add('Playing state', () => (
    <RecordingTimer
      timerSeconds="0:00"
      isRecording={false}
      status="Audio playing"
    />
  ))
  .add('Stopped state', () => (
    <RecordingTimer
      timerSeconds="0:00"
      isRecording={false}
      status="Audio stopped"
    />
  ));
