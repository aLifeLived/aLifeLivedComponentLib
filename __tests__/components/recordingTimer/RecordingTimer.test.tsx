import React from 'react';
import { render } from '../../../jest/renderUtil';

// Component
import { RecordingTimer } from '../../../src/components/recordingTimer/RecordingTimer';

const setup = (timerSeconds: string, isRecording: boolean, status: string) => {
  const utils = render(
    <RecordingTimer
      timerSeconds={timerSeconds}
      isRecording={isRecording}
      status={status}
    />
  );

  return { ...utils };
};
describe('Components >> Recording timer', () => {
  describe('Loading state', () => {
    test('Should successfully render recording timer in loading state', () => {
      const { toJSON } = setup('0:00', false, 'Loading please wait...');
      expect(toJSON()).toMatchSnapshot();
    });

    test('Should successfully render loading status and time', () => {
      const { getByText } = setup('0:00', false, 'Loading please wait...');
      expect(getByText('Loading please wait...')).not.toBeNull();
      expect(getByText('0:00')).not.toBeNull();
    });
  });

  describe('Ready state', () => {
    test('Should successfully render recording timer in ready state', () => {
      const { toJSON } = setup('0:00', false, 'Ready');
      expect(toJSON()).toMatchSnapshot();
    });

    test('Should successfully render ready status and time', () => {
      const { getByText } = setup('0:00', false, 'Ready');
      expect(getByText('Ready')).not.toBeNull();
      expect(getByText('0:00')).not.toBeNull();
    });
  });

  describe('Buffering state', () => {
    test('Should successfully render recording timer in buffering state', () => {
      const { toJSON } = setup('0:00', false, 'Buffering...');
      expect(toJSON()).toMatchSnapshot();
    });

    test('Should successfully render buffering status and time', () => {
      const { getByText } = setup('0:00', false, 'Buffering...');
      expect(getByText('Buffering...')).not.toBeNull();
      expect(getByText('0:00')).not.toBeNull();
    });
  });

  describe('Recording state', () => {
    test('Should successfully render recording timer in recording state', () => {
      const { toJSON } = setup('0:00', true, 'Recording');
      expect(toJSON()).toMatchSnapshot();
    });

    test('Should successfully render recording status and time', () => {
      const { getByText } = setup('1:40', true, 'Recording');
      expect(getByText('Recording')).not.toBeNull();
      expect(getByText('1:40')).not.toBeNull();
    });
  });

  describe('Playing state', () => {
    test('Should successfully render recording timer in playing state', () => {
      const { toJSON } = setup('0:00', true, 'Audio Playing');
      expect(toJSON()).toMatchSnapshot();
    });

    test('Should successfully render playing status and time', () => {
      const { getByText } = setup('1:40', false, 'Audio playing');
      expect(getByText('Audio playing')).not.toBeNull();
      expect(getByText('1:40')).not.toBeNull();
    });
  });

  describe('Stopped state', () => {
    test('Should successfully render recording timer in stopped state', () => {
      const { toJSON } = setup('0:00', true, 'Audio playing');
      expect(toJSON()).toMatchSnapshot();
    });

    test('Should successfully render stopped status and time', () => {
      const { getByText } = setup('1:40', false, 'Audio stopped');
      expect(getByText('Audio stopped')).not.toBeNull();
      expect(getByText('1:40')).not.toBeNull();
    });
  });
});
