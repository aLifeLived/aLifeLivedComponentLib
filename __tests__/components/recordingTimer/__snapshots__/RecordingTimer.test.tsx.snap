// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components >> Recording timer Buffering state Should successfully render recording timer in buffering state 1`] = `
<View
  style={
    Array [
      Object {
        "display": "flex",
      },
      Array [
        Object {},
      ],
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "display": "flex",
          "textAlign": "center",
        },
      ]
    }
  >
    0:00
  </Text>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
        },
        Array [
          Object {},
        ],
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#282828",
            "display": "flex",
            "fontSize": 16,
            "lineHeight": 20,
            "textAlign": "center",
          },
        ]
      }
    >
      Buffering...
    </Text>
  </View>
</View>
`;

exports[`Components >> Recording timer Loading state Should successfully render recording timer in loading state 1`] = `
<View
  style={
    Array [
      Object {
        "display": "flex",
      },
      Array [
        Object {},
      ],
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "display": "flex",
          "textAlign": "center",
        },
      ]
    }
  >
    0:00
  </Text>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
        },
        Array [
          Object {},
        ],
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#282828",
            "display": "flex",
            "fontSize": 16,
            "lineHeight": 20,
            "textAlign": "center",
          },
        ]
      }
    >
      Loading please wait...
    </Text>
  </View>
</View>
`;

exports[`Components >> Recording timer Playing state Should successfully render recording timer in playing state 1`] = `
<View
  style={
    Array [
      Object {
        "display": "flex",
      },
      Array [
        Object {},
      ],
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "display": "flex",
          "textAlign": "center",
        },
      ]
    }
  >
    0:00
  </Text>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
        },
        Array [
          Object {},
        ],
      ]
    }
  >
    <View
      forwardedRef={[Function]}
      style={
        Object {
          "opacity": 0,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#D94E69",
              "borderRadius": 8,
              "display": "flex",
              "height": 10,
              "marginRight": 8,
              "width": 10,
            },
            Array [
              Object {},
            ],
          ]
        }
      />
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#282828",
            "display": "flex",
            "fontSize": 16,
            "lineHeight": 20,
            "textAlign": "center",
          },
        ]
      }
    >
      Audio Playing
    </Text>
  </View>
</View>
`;

exports[`Components >> Recording timer Ready state Should successfully render recording timer in ready state 1`] = `
<View
  style={
    Array [
      Object {
        "display": "flex",
      },
      Array [
        Object {},
      ],
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "display": "flex",
          "textAlign": "center",
        },
      ]
    }
  >
    0:00
  </Text>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
        },
        Array [
          Object {},
        ],
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#282828",
            "display": "flex",
            "fontSize": 16,
            "lineHeight": 20,
            "textAlign": "center",
          },
        ]
      }
    >
      Ready
    </Text>
  </View>
</View>
`;

exports[`Components >> Recording timer Recording state Should successfully render recording timer in recording state 1`] = `
<View
  style={
    Array [
      Object {
        "display": "flex",
      },
      Array [
        Object {},
      ],
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "display": "flex",
          "textAlign": "center",
        },
      ]
    }
  >
    0:00
  </Text>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
        },
        Array [
          Object {},
        ],
      ]
    }
  >
    <View
      forwardedRef={[Function]}
      style={
        Object {
          "opacity": 0,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#D94E69",
              "borderRadius": 8,
              "display": "flex",
              "height": 10,
              "marginRight": 8,
              "width": 10,
            },
            Array [
              Object {},
            ],
          ]
        }
      />
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#282828",
            "display": "flex",
            "fontSize": 16,
            "lineHeight": 20,
            "textAlign": "center",
          },
        ]
      }
    >
      Recording
    </Text>
  </View>
</View>
`;

exports[`Components >> Recording timer Stopped state Should successfully render recording timer in stopped state 1`] = `
<View
  style={
    Array [
      Object {
        "display": "flex",
      },
      Array [
        Object {},
      ],
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "display": "flex",
          "textAlign": "center",
        },
      ]
    }
  >
    0:00
  </Text>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
        },
        Array [
          Object {},
        ],
      ]
    }
  >
    <View
      forwardedRef={[Function]}
      style={
        Object {
          "opacity": 0,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#D94E69",
              "borderRadius": 8,
              "display": "flex",
              "height": 10,
              "marginRight": 8,
              "width": 10,
            },
            Array [
              Object {},
            ],
          ]
        }
      />
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#282828",
            "display": "flex",
            "fontSize": 16,
            "lineHeight": 20,
            "textAlign": "center",
          },
        ]
      }
    >
      Audio playing
    </Text>
  </View>
</View>
`;
